.github/workflows/tests.yaml

```diff
@@ -46,3 +55,27 @@ jobs:
 
       - name: Check resources
         run: kubectl get all -n checkpoint-restore-operator-system
+
+      - name: Test Garbage Collection
+        run: |
+          ls -la ${{ env.CHECKPOINT_DIR }}
+          kubectl apply -f ./test/test_checkpointrestoreoperator.yaml
+          ./test/wait_for_checkpoint_reduction.sh 2
+          ls -la ${{ env.CHECKPOINT_DIR }}
+
+      - name: Test Max Checkpoints Set to 0
+        run: |
+          sed -i 's/maxCheckpointsPerContainer: [0-9]*/maxCheckpointsPerContainer: 0/' ./test/test_checkpointrestoreoperator.yaml
+          sed -i '/^  containerPolicies:/,/maxCheckpoints: [0-9]*/ s/^/#/' ./test/test_checkpointrestoreoperator.yaml
+          kubectl apply -f ./test/test_checkpointrestoreoperator.yaml
+          ./test/generate_checkpoint_tar.sh
+          sleep 2
+          ./test/wait_for_checkpoint_reduction.sh 7
+          ls -la ${{ env.CHECKPOINT_DIR }}
```

Comment:
This test workflow itself can be a shell script that we invoke here - it would probably be cleaner than repeating all of this in the block below.

Reply:
@snprajwal I added a test script that includes different functions for various edge cases because if we kept separate scripts for all tests, there would be too many scripts later on.